src/Uno.UI/UI/Xaml/Controls/ContentDialog/DeferralManager.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Windows.UI.Xaml.Controls
+{
+	/// <summary>
+	/// Handles completion of deferrals. The deferred action is completed when all deferral objects that were taken have called Complete().
+	/// </summary>
+	/// <typeparam name="T"></typeparam>
+	internal class DeferralManager<T> where T : IDeferral, new()
+	{
+		private readonly Action _completeHandler;
+		private int _deferralsCount;
+
+		public DeferralManager(Action completeHandler)
+		{
+			this._completeHandler = completeHandler;
+		}
+
+		public T GetDeferral()
+		{
+			_deferralsCount++;
```

Comment:
What if we try to `GetDefferal` after a previous one was completed? (so the `_completeHandler()` will already have been invoked)

Reply:
ooh good point

Reply:
What does WinUI do ?

Reply:
So for this case:
```csharp
			dialog.Closing += (o, e) =>
			{
				deferral1 = e.GetDeferral();
				deferral1.Complete();
			};
			dialog.Closing += (o, e) =>
			{
				deferral2 = e.GetDeferral();
			};
```

It won't close the dialog until `deferral2.Complete()` is eventually called. Making the change now.

Reply:
fixed